locator/tablet_sharder.hh

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "locator/tablets.hh"
+#include "dht/sharder.hh"
+#include "locator/token_metadata.hh"
+
+namespace locator {
+
+class tablet_sharder : public dht::sharder {
+    const token_metadata& _tm;
+    table_id _table;
+    mutable const tablet_map* _tmap = nullptr;
+private:
+    void ensure_tablet_map() const {
+        if (!_tmap) {
+            _tmap = &_tm.tablets().get_tablet_map(_table);
+        }
+    }
+public:
+    tablet_sharder(const token_metadata& tm, table_id table)
+            : _tm(tm)
+            , _table(table)
+    { }
```

Comment:
Shouldn't the lookup be done here?

Reply:
tablet_sharder is instantiated with erm. In some cases we instantiate erm with token_metadata which is lacking info for a gicven table, temporarily. I would have to swallow the lookup error and use optionals to return the error in accessors.